### ISSUE TITLE
Don't require ActorMaterializerSettings to be used for new settings

### ISSUE THREAD

**Issue by jrudolph** [akka/akka#27992]
The current ActorMaterializerSettings is now deprecated but it is still an essential part in providing default values for mandatory attributes in streams. For binary compatibility reasons it contains a huge amount of hacks and weird constructors which makes it hard and error-prone to add new settings.

It would be good if we could retire `ActorMaterializerSettings` for any actual (internal or other) usage in 2.6 and already provide an alternative (internal?) structure to carry the same information that was previously provided by `ActorMaterializerSettings`.

**Comment by johanandren** [comment]
The problem with completely retiring it is that it breaks existing (deprecatic) programmatic config of defaults, so even if it would be nice if we didn't have to maintain it I think we have to keep it. 

I suggest we start on a new `StreamSettings` class that is Akka-internal and follow our usual pattern for loading settings for a module, to load any new settings and then when we can throw away the `ActorMaterializerSettings` we move the existing settings from there into the internal class.

**Comment by jrudolph** [comment]
Yes, sorry, I left "retiring" open to interpretation. I meant we still keep it around and happy for the time being but make sure it doesn't get any further tasks until it fades away and we promote it to glory, so it can hop on the last rattler to cross the rainbow bridge and join the whisperers.
